### ISSUE TITLE
lint errors with 4.8 but not with 4.7.7

### ISSUE THREAD

**Issue by japrescott** [garronej/tss-react#167]
Hey @garronej !

Thanks a lot for the package, great stuff!

I wanted to upgrade tss-react but noticed that with 4.8.x I am getting linting errors
```
> tsc --noEmit && eslint --color -c ./.eslintrc.commit.js './src/**/*.{ts,tsx}'

../../node_modules/tss-react/mui.d.ts:34:124 - error TS1005: '?' expected.

34     }) => Classes extends Record<string, unknown> ? Classes extends Partial<Record<infer K extends string | number | symbol, any>> ? Record<K, string> : Classes : {
```

tss-react 4.7.7 works without an issue. I also had a look at your recent changes but can't spot anything which would have caused this to my knowlege.

eslint: 8.39.0
typescript: 4.4.4
target: ES2020

**Comment by garronej** [comment]
> Thanks a lot for the package, great stuff!  

Thank you for the kind words!  

I have to investigate but do you think it might be related to: https://github.com/garronej/tss-react/issues/158?

**Comment by garronej** [comment]
I see what's hapenning, fixing...

**Comment by garronej** [comment]
Should be fixed in 4.8.3  

Thanks for reporting!
